### PR TITLE
Feat/sign in with email

### PR DESCRIPTION
**Pull request by pitb2022**

### CODE REVIEW

File: src/components/Forms/EmailLoginForm/index.tsx [hunk 1]
```diff
@@ -131,10 +167,18 @@ export const EmailLoginForm: React.FC<FormProps> = ({
...
-            setFieldError('email', messages[code])
-          } else if (code.indexOf('USER_PASSWORD_') >= 0) {
+          if (
+            code.includes('USER_EMAIL_') ||
+            code.indexOf('USER_PASSWORD_') >= 0 ||
+            code.includes('CODE_INVALID') ||
+            code.includes('CODE_INACTIVE')
+          ) {
+            const m = translate({ id: 'USER_PASSWORD_INVALID', lang })
+            setFieldError('email', m)
```
Comment (robertu7) [inline src/components/Forms/EmailLoginForm/index.tsx:179]: please confirm if showing the same error message on both email and password fields is correct.

File: src/components/Forms/EmailLoginForm/index.tsx [hunk 2]
```diff
@@ -145,6 +189,31 @@ export const EmailLoginForm: React.FC<FormProps> = ({
...
+    const error = validateEmail(values.email, lang, { allowPlusSign: true })
+    if (error) {
+      setFieldError('email', error)
+      return
+    }
+
+    await sendCode({
+      variables: { input: { email: values.email, type: 'email_otp' } },
+    })
+    setCountdown(10)
```
Comment (robertu7) [inline src/components/Forms/EmailLoginForm/index.tsx:202]: 60s?

File: src/components/Forms/EmailLoginForm/index.tsx [hunk 3]
```diff
@@ -66,19 +76,42 @@ export const USER_LOGIN = gql`
...
+  const isWallet = authTypeFeed === 'wallet'
+
+  const [isSelectMethod, setIsSelectMethod] = useState(false)
+  const [errorCode, setErrorCode] = useState<any>(null)
+  const [hasSendCode, setHasSendCode] = useState(false)
+  const [countdown, setCountdown] = useState(0)
+
+  const passwordRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
```
Comment (robertu7) [inline src/components/Forms/EmailLoginForm/index.tsx:103]: can we reuse the `useCountdown` hook (also used by `src/components/Forms/Verification/SendCodeButton/index.tsx`)

File: src/components/Form/Input/index.tsx [hunk 4]
```diff
@@ -22,72 +23,110 @@ import styles from './styles.module.css'
...
+                name={name}
+                type={type}
+                aria-describedby={fieldMsgId}
+                autoComplete="nope"
+                autoCorrect="off"
+                autoCapitalize="off"
+                spellCheck="false"
+                // className={inputClasses}
+              />
+              {rightButton}
```
Comment (robertu7) [inline src/components/Form/Input/index.tsx:93]: need some space between input and `rightButton`

image

File: src/components/Form/Input/index.tsx [hunk 5]
```diff
@@ -22,72 +23,110 @@ import styles from './styles.module.css'
...
+            </section>
+          </Field.Content>
+        )}
+        {!rightButton && (
```
Comment (robertu7) [inline src/components/Form/Input/index.tsx:97]: Can we combine `right &&` and `!right &&` conditional statements (keep the L78-L95 and remove L97-L113)?

**Review by robertu7 — CHANGES_REQUESTED** [review]

File: src/components/Forms/EmailLoginForm/index.tsx (hunk 1 above)
Comment (pitb2022) [reply src/components/Forms/EmailLoginForm/index.tsx:180]: yes

**Review by robertu7 — APPROVED** [review]

**Pull request merged**